app/controllers/solidus_bolt/transactions_controller.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  class TransactionsController < ::Spree::Api::BaseController
+    def authorize
+      order = ::Spree::Order.find_by(number: params[:order_id])
+      response = ::SolidusBolt::Transactions::AuthorizeService.call(
+        order: order, create_bolt_account: params[:create_bolt_account], credit_card: credit_card_params
+      )
+      status = response.success? ? :ok : :internal_server_error
+
+      render json: response, status: status
+    end
+
+    private
+
+    def credit_card_params
+      params[:credit_card].permit(:token, :bin, :last4, :token_type, :expiration, :network)
+    end
```

Comment:
Do we need that action? 
I think we don't need it.
Providing the right data to [Update checkout API](https://solidus.stoplight.io/docs/solidus/b3A6NDU3NzY5NDE-update-checkout), it will create the Spree::Payment and related source under the hood. It will just require to implement the SolidusBolt::Gateway#authorize method.

Reply:
Maybe the [issue](https://github.com/nebulab/solidus_bolt/issues/30) was not well defined.
I have updated it to reflect what should are the goal for this PR.

Reply:
hmm, I see what you are trying to say

I'm now trying to rearrange based on that, but we have an issue: the payment source stores values that we get only after calling the Bolt Authorize API. Will try to figure out how to rearrange it, but we might need to jump on a call